docs/pages/collection/[collectionAddress].tsx

```diff
@@ -0,0 +1,253 @@
+  data: NftokenTypes.NftInfo[];
+  error: any;
+  mutate: SWRResponse<NftokenTypes.NftInfo[], never>["mutate"];
+} => {
+  const swrKey = [collectionAddress, network, "getNftsInCollection"];
+  const { data, error, mutate } = useSWR(swrKey, async () => {
+    return await NftokenFetcher.getNftsInCollection({
+      collection: collectionAddress,
+      network,
+    });
+  });
+  return { data: data!, error, mutate };
+};
+
+const KEYS: (keyof NftokenTypes.CollectionInfo)[] = [
+  "address",
+  "authority",
+  "authority_can_update",
+  "description",
+  "image",
+  "minted_at",
+  "metadata_url",
+];
+
+export default function CollectionPage({
+  initialCollection,
+  initialNftsInCollection,
+}: {
+  initialCollection: NftokenTypes.CollectionInfo;
+  initialNftsInCollection: NftokenTypes.NftInfo[];
+}) {
+  const {
+    query: { collectionAddress, network },
+  } = useRouter();
+
+  const { data: collectionData } = useCollection({
+    collectionAddress: collectionAddress as string,
+    network: network ? (network as Network) : Network.Mainnet,
+  });
+
+  const { data: collectionNftsData } = useCollectionNfts({
+    collectionAddress: collectionAddress as string,
+    network: network ? (network as Network) : Network.Mainnet,
+  });
+
+  const collection = collectionData ?? initialCollection;
+  const nftsInCollection = collectionNftsData ?? initialNftsInCollection;
+
+  if (collection === undefined) {
+    return <div>Loading...</div>;
+  }
+
+  if (collection === null) {
+    return (
+      <>
+        <SocialHead subtitle="NFT Not Found" />
+        <h1>We couldn’t find a collection with this address.</h1>
+        <style jsx>{`
+          h1 {
+            font-size: 1.5rem;
+            padding: 1rem;
+          }
+        `}</style>
+      </>
+    );
+  }
+
+  const attributes: { [key: string]: any } = {};
+  for (const key of KEYS) {
+    attributes[key] = collection[key];
+  }
+
+  const traits: { [key: string]: any } = {};
+  if (collection.traits) {
+    for (const entry of collection.traits) {
+      traits[entry.trait_type] = entry.value;
+    }
+  }
+
+  return (
+    <PageLayout>
+      <SocialHead subtitle={collection.name} />
+      <div className="columns">
+        <div>
+          <div className="collection-badge">Collection</div>
+          <h1>{collection.name}</h1>
+          <ValueList attributes={attributes} />
+        </div>
+        <div className="traits-column">
+          <h2>Traits</h2>
+          {collection.traits && collection.traits.length > 0 ? (
+            <ValueList attributes={traits} />
+          ) : (
+            <div className="traits-empty-state">No traits set.</div>
+          )}
+        </div>
+      </div>
+
+      {nftsInCollection && (
+        <div className="nft-container mt-5">
+          {nftsInCollection.map((nft) => (
+            <a href={`/nft/${nft.address}`} key={nft.address} className="nft">
+              <img src={nft.image} alt={nft.name} className="nft" />
```

Comment:
Whenever we use an image we should use `getImageUrl` since then we are using our image CDN. We should also create a component `<SquareImage />` that uses the CDN